### PR TITLE
109-editdelete-posting-to-be-dynamically

### PR DESCRIPTION
**Pull request by khaled106**
The posting page is now fully functional as the employer is able to Add and delete postings since the merged PR #112. Here we implemented edit postings as the user can now edit existing positions.

### CODE REVIEW

**Review by mario-elshaer — APPROVED** [review]
Great work on fixing the modal such that we can now edit or add depending on the need. Code looks neat! Great work all around.

File: client/src/components/Postings/EditPosting.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    const url = `http://localhost:3001/postings/${id}`;
+    const req = {
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        // no need for them to be sent keep them in case we need to change that later.
+        // employerID: userInfo.id,
+        // organizationName: userInfo.organizationName,
```
Comment (ramy-att) [inline client/src/components/Postings/EditPosting.js:67]: If the above comments are not needed you can remove them

File: client/src/components/Postings/EditPosting.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        expirationDate: expirationDate,
+        location: location,
+      }),
+    };
+    try {
+      const response = await fetch(url, req);
+      const result = await response.json();
+      console.log(result);
+      if (!result.err) {
+        // history.push("/job-postings");
```
Comment (ramy-att) [inline client/src/components/Postings/EditPosting.js:79]: Same here

File: client/src/components/Postings/EditPosting.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+              <option value="Hybrid">Hybrid</option>
+              <option value="In-Person">In-Person</option>
+              <option value="custom">Custom</option>
+            </Form.Control>
+          </Form.Group>
+          {location === "custom" && (
+            <Form.Group controlId="formFile" className="mb-3">
+              <Form.Label>Location*</Form.Label>
+              <Form.Control required placeholder="Montreal" />
+            </Form.Group>
```
Comment (ramy-att) [inline client/src/components/Postings/EditPosting.js:150]: @khaled106 can you please state-control this FormControl, currently if the user sets a custom location it appears as this:
image
We would like to set the location to the custom entered location

File: client/src/components/Postings/Posting.js [hunk 4]
```diff
@@ -213,56 +213,16 @@ const Posting = () => {
   return (
     <Container fluid className="postings-page">
       <div className="title-container">
-        <h1>CAE: Front-End Intern (ID: #{id})</h1>
+        <h1>
+          {" "}
+          {posting.organizationName}: {posting.title}
+        </h1>
```
Comment (ramy-att) [inline client/src/components/Postings/Posting.js:219]: Nice

**Review by ramy-att — CHANGES_REQUESTED** [review]
Nicely done! Great job! Just left a couple of small inline comments.

**Review by ramy-att — APPROVED** [review]

**Pull request merged**